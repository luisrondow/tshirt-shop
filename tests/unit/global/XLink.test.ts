import { describe, it, expect } from 'vitest'
import { mount, shallowMount } from '@vue/test-utils'

import { XLink } from '#components'

describe('XLink component', () => {
  it('should render correctly', () => {
    const wrapper = mount(XLink)

    expect(wrapper.vm).toBeTruthy()
    expect(wrapper.element).toMatchSnapshot()
  })

  it('should render correctly with "to" props', () => {
    const wrapper = shallowMount(XLink, {
      props: {
        to: '/test',
      },
    })

    expect(wrapper.find('nuxt-link-stub').attributes('href')).toBe('/test')
  })

  it('should render correctly with "href" props', () => {
    const wrapper = shallowMount(XLink, {
      props: {
        href: '/test',
      },
    })

    expect(wrapper.find('a').attributes('href')).toBe('/test')
  })

  it('should render span without props', () => {
    const wrapper = shallowMount(XLink)

    expect(wrapper.find('nuxt-link-stub').exists()).toBeFalsy()
    expect(wrapper.find('a').exists()).toBeFalsy()
    expect(wrapper.find('span').exists()).toBeTruthy()
  })
})
