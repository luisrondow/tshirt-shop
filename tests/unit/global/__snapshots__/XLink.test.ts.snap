// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`XLink component > should render correctly 1`] = `
<span>
  
  
</span>
`;
